NebulaHost/PacketProcessors/Factory/Entity/CreatePrebuildsRequestProcessor.cs

```diff
@@ -48,61 +48,65 @@ public void ProcessPacket(CreatePrebuildsRequest packet, NebulaConnection conn)
-                    AccessTools.Field(typeof(PlayerAction_Build), "factory").SetValue(GameMain.mainPlayer.controller.actionBuild, tmpFactory);
-                    AccessTools.Field(typeof(PlayerAction_Build), "nearcdLogic").SetValue(GameMain.mainPlayer.controller.actionBuild, tmpNearcdLogic);
+                pab.waitConfirm = true;
+                using (FactoryManager.EventFromServer.On())
+                {
+                    FactoryManager.EventFactory = planet.factory;
+                    pab.previewPose.position = new UnityEngine.Vector3(packet.PosePosition.x, packet.PosePosition.y, packet.PosePosition.z);
+                    pab.previewPose.rotation = new UnityEngine.Quaternion(packet.PoseRotation.x, packet.PoseRotation.y, packet.PoseRotation.z, packet.PoseRotation.w);
+
+                    //Check if some mandatory variables are missing
+                    if (planet.physics == null || planet.physics.colChunks == null)
+                    {
+                        planet.physics = new PlanetPhysics(planet);
+                        planet.physics.Init();
+                    }
+                    if (AccessTools.Field(typeof(CargoTraffic), "beltRenderingBatch").GetValue(planet.factory.cargoTraffic) == null)
+                    {
+                        planet.factory.cargoTraffic.CreateRenderingBatches();
+                    }
+                    if (planet.aux == null)
+                    {
+                        planet.aux = new PlanetAuxData(planet);
+                    }
+
+                    //Set temporary Local Planet / Factory data that are needed for original methods CheckBuildConditions() and CreatePrebuilds()
+                    AccessTools.Field(typeof(PlayerAction_Build), "factory").SetValue(GameMain.mainPlayer.controller.actionBuild, planet.factory);
+                    AccessTools.Field(typeof(PlayerAction_Build), "planetPhysics").SetValue(GameMain.mainPlayer.controller.actionBuild, planet.physics);
+                    AccessTools.Field(typeof(PlayerAction_Build), "nearcdLogic").SetValue(GameMain.mainPlayer.controller.actionBuild, planet.physics.nearColliderLogic);
+                    AccessTools.Property(typeof(global::Player), "planetData").SetValue(GameMain.mainPlayer, planet, null);
+
+                    //Check if prebuilds can be build (collision check, height check, etc)
+                    GameMain.mainPlayer.mecha.buildArea = float.MaxValue;
+                    bool canBuild;
+                    using (FactoryManager.IgnoreBasicBuildConditionChecks.On())
+                    {
+                        canBuild = pab.CheckBuildConditions();
+                        canBuild &= CheckBuildingConnections(pab.buildPreviews, planet.factory.entityPool, planet.factory.prebuildPool);
+                    }
+
+                    if (canBuild)
+                    {
+                        FactoryManager.PacketAuthor = packet.AuthorId;

```

Comment:
I'm thinking this could be covered by a scope, but preferred not to mess with it. You can basically have a scope for this value too.